modin/numpy/arr.py

```diff
@@ -165,7 +176,11 @@ def __init__(
                     target_kwargs.pop(key)
                 else:
                     target_kwargs[key] = locals()[key]
-            arr = numpy.array(object, **target_kwargs)
+            if not isinstance(object, numpy.ndarray):
+                arr = numpy.array(object, **target_kwargs)
+            else:
+                # Avoid creating a second NumPy array in memory before distribution.
+                arr = object
```

Comment:
```suggestion
            arr = numpy.asarray(object, **target_kwargs)
```
should do what you want - avoid copying if `object` is already an array of desired type etc., see [numpy docs](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html)

Reply:
Fixed!